# Get Vasari letters from Fondazione Memofonte (https://www.memofonte.it/ricerche/giorgio-vasari/) with Beautiful Soup

#### based on Jeri Wieringa, "Intro to Beautiful Soup," The Programming Historian 1 (2012), https://doi.org/10.46430/phen0008.

In [1]:
import urllib.request, urllib.error, urllib.parse
import csv
from bs4 import BeautifulSoup
import html2text
import os
from datetime import datetime
import numpy as np
import pandas as pd

In [2]:
# function to check if 'nan'

def isnan(x):
    return x != x
    

In [3]:
data_path = 'data/results/letters/'

## Iterate over all URLs and save as .html

#### make list of URLs

In [2]:
# letter-id ranging from 1 to 1151

url_list = []
url_base = "https://www.memofonte.it/home/ricerca/singolo_17.php?id="
for i in range (1151):
    url_list.append(url_base+ str(i+1))
    
len(url_list)

1151

### save all as .html

In [40]:
count = 0
for i in url_list:
    count+=1
    response = urllib.request.urlopen(i)
    webContent = response.read()
    #filename = 'data/results/letters/html/letter-'+str(i)'.html'
    f = open('data/results/letters/html/letter-'+str(count)+'.html', 'wb')
    f.write(webContent)
    f.close()
    

## Open all .html, make Beautiful Soups, extract information, save all letters in a single CSV

In [6]:
list_of_htmls = os.listdir('data/results/Vasari_letters_csv/html')
len(list_of_htmls)
#list_of_htmls

1151

In [7]:
# obo.py
# get rid of tags without Beautiful Soup
# because there is a space " " missing after some <br/> tags

def stripTags(pageContent):
    pageContents = str(pageContent)

    inside = 0
    text = ''
    storage = " "

    for char in pageContents:
        if (char == '<' and storage == " "):
            inside = 1
        elif (char == '<' and storage != " "):
            inside = 1
            text += " "     
        elif (inside == 1 and char == '>'):
            inside = 0
            #text += " "
        elif inside == 1:
            continue
        else:
            text += char
        storage = char

    return text

In [34]:
# inspect data
#soup.prettify

### Make one CSV File for all letters

In [129]:
# make CSV File
# changed filename after first run from letters.csv to letters_x.csv
csv_file = csv.writer(open('data/results/Vasari_letters_csv/letters_x.csv', "w"))

# Write column headers as the first line
csv_file.writerow(["Numero d'ordine", "Data", "Intestazione", "Segnatura", "Fonte", "Bibliografia"]) 

64

#### iterate over all .html files, save as csv

In [130]:
# open all letter.html, make soup, extract information, collect it in a list, append list as a row to csv_file

for i in list_of_htmls:
    csv_row = list() # make new row
    soup = BeautifulSoup(open('data/results/letters/html/'+i), "lxml") # make soup
    p_array = soup.find_all('p') # take the information one by one 
    csv_row.append(int(p_array[0].text[(len('Numero d\'ordine:')+1):])) # append each to the list
    csv_row.append(p_array[1].text[(len('Data:')+1):])
    csv_row.append(p_array[2].text[(len('Intestazione:')+1):])
    csv_row.append(p_array[3].text[(len('Segnatura:')+1):])
    #csv_row.append(p_array[4].text[(len('Fonte:')+1):])
    x = (p_array[4])
    formatted_text = stripTags(str(x)[12:]) # remove html tags without beautiful soup
    csv_row.append(formatted_text.replace("\xa0", ''))
    
    csv_row.append(p_array[5].text[(len('Bibliografia:')+1):])
    csv_file.writerow(csv_row)

## open or save CSV-file as dataframe

In [47]:
#filename = 'data/results/Vasari_letters_csv/letters_2.csv'
filename = 'data/results/Vasari_letters_csv/letters_4.csv'

dataframe = pd.read_csv(filename) # open csv
#dataframe.to_csv(filename, index = False, header = True) # save as csv

dataframe["Fonte"][1146]

'Reverendissimo monsignore patron mio, io mando alla Signoria Vostra Reverendissima questa inclusa ch’è della Comunità d’Arezzo; la Signoria Vostra sia contenta, spedita ch’ella sarà, di rimandamela in mano, perché importa alla mia Comunità e ve ne arò obligo. Io vi ho scritto e Sua Eccellenzia doverrà deliberare quanto gli piacerà delle cose che bosogniano per qua per servizio suo. Ebbi un’ultima sua col sugel grosso, farò quanto m’è comesso e tuttavia si atende; così potessi voi con tutta la corte goder quel ch’è fatto senza avere a star ne’ luoghi che son disfatti, pure il faro che vi conduce lo debbe saper lui; andrò pregando Dio che vi riconduca a noi e che vi dia pazienzia che questo santo ch’è oggi nel giorno, che io scrivo, i diavoli gniene scossono coi bastoni; dove il nostro Bernia dice in questi due versi: a giorni diciassette l’Alcionio fu bastonato come santo Antonio. E con questo fo fine, restando al suo comando. Di vostra Signoria reverendissimo servitore Giorgio Vasari 

## get rid of nan-values

'nan' count in total: 11 rows with nan values 
--> replace with '0' and collect index for documentation (row-indices found: 17, 101, 268, 368, 491, 569, 617, 703, 740, 928, 985)

In [9]:
count = 0
for i in range(len(dataframe)): # check column by column if isnan and replace with '0'
    
    if isnan(dataframe['Intestazione'][i]): 
        print('Intestazione, ', count)
        (dataframe['Intestazione'][i]) = '0'
        
    if isnan(dataframe['Segnatura'][i]):
        #print('Segnatura', count)
        (dataframe['Segnatura'][i]) = '0'
        
    if isnan(dataframe['Fonte'][i]):
        #print('Fonte', count)
        (dataframe['Fonte'][i]) = '0'
        
    if isnan(dataframe['Bibliografia'][i]):
        #print('Bibliografia', count)
        (dataframe['Bibliografia'][i]) = '0'
    
    count+=1

Intestazione,  17
Intestazione,  101
Intestazione,  268
Intestazione,  368
Intestazione,  491
Intestazione,  569
Intestazione,  617
Intestazione,  703
Intestazione,  740
Intestazione,  928
Intestazione,  985


<ipython-input-9-f6e2de138291>:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  (dataframe['Intestazione'][i]) = '0'
<ipython-input-9-f6e2de138291>:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  (dataframe['Segnatura'][i]) = '0'
<ipython-input-9-f6e2de138291>:18: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  (dataframe['Bibliografia'][i]) = '0'


# Distinguish letters by Vasari himself, copied by his nefew and authorship candidates

example for signature of letter copied by nefew (called Giorgio Vasari the younger) 
BRF, 2354, cc. 12v-14, copia di Giorgio Vasari il Giovane.

number of letters filtered at http://vasariscrittore.memofonte.it/archivio
390

In [ ]:
filename = 'data/results/Vasari_letters_csv/letters_4.csv'
#dataframe = pd.read_csv(filename) # open csv

In [73]:
dataframe['Intestazione'] = [i.strip(' ') for i in dataframe['Intestazione']]

In [31]:
# check who has sent how many letters
sender = df_labeled['Intestazione']
store = []
for i in sender:
    store.append(i[:15])    
for i in set(store):
    print(i, store.count(i))

RETTORI E DEPUT 2
JACOPO APPIANI  1
JACOPO GUIDI IN 5
CRISTOFANO DI M 5
FEDERICO DI MON 12
AVERARDO SERRIS 1
BARTOLOMEO CONC 14
LELIO BONSI IN  2
GIORGIO VASARI  404
VINCENZO BORGHI 136
DANIELE RICCIAR 2
ANTONIO SERGUID 1
0 11
GABRIELE BOMBAS 1
TIBERIO CALCAGN 1
FRANCESCO DE’ M 7
GIOVANNI DI MON 1
ALESSANDRO FARN 1
GIROLAMO RUSTIC 1
LEONARDO BUONAR 1
VESCOVO SARACIN 1
BENEDETTO LOMEL 3
FRANCESCO DI SE 1
PIO V A GIORGIO 1
SPINELLO BENCI  3
GABRIELE PALEOT 1
PIER VETTORI IN 1
PRIORE DELL’ORA 1
BATTISTA LORENZ 1
STEFANO VELTRON 1
MARCHESE DI CET 1
GABRIEL FIAMMA  16
PROVEDITORI DE’ 2
GIOVAN MARIA DE 1
FILIPPO BUONCOM 14
ALESSANDRO MUSO 2
CHIAPPINO VITEL 1
PIETRO ARETINO  10
VESCOVO CORNARO 1
FRA’ STEFANO IN 2
DON ANTONIO IN  1
PIERGIOVANNI AL 1
COSIMO DE’ MEDI 43
GIOVANNI BATTIS 12
TOLOMEO GALLI I 2
ORDINE DI SANTO 10
ALESSANDRO STRO 2
PIERFRANCESCO G 6
MARCO SARACINI  1
ANONIMO DA PRAT 1
MINIATO PITTI I 23
LEONARDO MARINO 1
COSIMO BARTOLI  73
GIULIO RICASOLI 1
LUDOVICO CESI I 6
GUIDO FER

- BERNARDETTO MIN 66
- GUGLIELMO SANGA 108
- GIORGIO VASARI  404
- VINCENZO BORGHI 136
- COSIMO BARTOLI  73

### split and label dataframe of letters for different involved authorship candidates

Cosimo Bartoli and Pierfrancesco Giambullari involved as Co-Authors after Gerd Blum (https://archiv.ub.uni-heidelberg.de/artdok/1948/1/Blum_Provvidenza_e_progresso_2010.pdf) 

Pierfrancesco Giambullari, Cosimo Bartoli, Giovan Battista Gelli und Carlo Lenzoni involved in the preparation of the edition of 1550; massive contribution of Vincenzio Borghinis concerning the indices and growing involvement over the following decennies (Alessandro Nova: Vasari versus Vasari, S.151)

In [74]:
# get Pierfrancesco Giambullari, Cosimo Bartoli, Vincenzio Borghini, Minerbetti Bernadetto, Sangaletti Guglielmo

# examples of the signatures copied from http://vasariscrittore.memofonte.it/archivio# 
# COSIMO BARTOLI IN FIRENZE A GIORGIO VASARI IN ROMA (73 letters)
# VINCENZO BORGHINI IN FIRENZE A GIORGIO VASARI IN ROMA (131 letters)
# PIERFRANCESCO GIAMBULLARI IN FIRENZE A GIORGIO VASARI IN AREZZO (6 letters)
# MINERBETTI BERNARDETTO 108 letters
# SANGALLETTI GUGLIELMo 66 letters
# GIORGIO VASARI 363
# GIORGIO VASARI copied by his nephew 41



Borghini = list() # collect whole entries written by Vincenzo Borghini
Giambullari = list() # collect Pierfrancesco Giambullari
Bartoli = list() # collect Cosimo Bartoli
Minerbetti = list() # collect MINERBETTI BERNARDETTO
Sanga = list()  # collect SANGALLETTI GUGLIELMO
Vasari = list() # Giorgio Vasari 363
Vasari_c = list() # Giorgio Vasari copies 41

new_column = list()

for i in range(len(dataframe)):

    if dataframe['Intestazione'][i].startswith('VINCENZO BORGHINI'): # VINCENZO BORGHINI
        Borghini.append(list(dataframe.iloc[i]))
        new_column.append('Borghini')
    
    elif 'COSIMO BARTOLI' in dataframe['Intestazione'][i]: # COSIMO BARTOLI
        Bartoli.append(list(dataframe.iloc[i]))
        new_column.append('Bartoli')
    
    elif 'PIERFRANCESCO GIAMBULLARI' in dataframe['Intestazione'][i]: # Pierfrancesco Giambullari
        Giambullari.append(list(dataframe.iloc[i]))
        new_column.append('Giambullari')
        
    elif dataframe['Intestazione'][i].startswith('BERNARDETTO MIN'): # BERNADETTO MINERBETTI
        Minerbetti.append(list(dataframe.iloc[i]))
        new_column.append('Minerbetti')
        
    elif dataframe['Intestazione'][i].startswith('GUGLIELMO SANGA'): # VINCENZO BORGHINI
        Sanga.append(list(dataframe.iloc[i]))
        new_column.append('Sanga')
        
    elif dataframe['Intestazione'][i].startswith('GIORGIO VASARI') and 'copia' in dataframe['Segnatura'][i]: # VINCENZO BORGHINI
        Vasari_c.append(list(dataframe.iloc[i]))
        new_column.append('Vasari_c')
    
    elif dataframe['Intestazione'][i].startswith('GIORGIO VASARI') and 'copia' not in dataframe['Segnatura'][i]: # filter out copies        
        Vasari.append(list(dataframe.iloc[i]))
        new_column.append('Vasari')
        
    else:
        new_column.append('other')
        
dataframe['author'] = new_column
print('Giambullari: ', len(Giambullari), 'Borghini: ', len(Borghini), 'Bartoli: ', len(Bartoli), 'Sanga: ', len(Sanga), 'Minerbetti: ', len(Minerbetti), 'Vasari: ', len(Vasari), 'Vasari_c: ', len(Vasari_c))

Giambullari:  6 Borghini:  136 Bartoli:  73 Sanga:  108 Minerbetti:  66 Vasari:  363 Vasari_c:  41


In [55]:
columns = ["Numero d'ordine", "Data", "Intestazione", "Segnatura", "Fonte", "Bibliografia", "Label", "author" ]
data_path = 'data/results/Vasari_letters_csv/'

In [75]:
dataframe

Unnamed: 0  Numero d'ordine         Data  \
0              0             1051  26 12  1572   
1              1              791  24 12  1569   
2              2              284  20 04  1559   
3              3             1114   1 08  1573   
4              4              904   3 09  1571   
...          ...              ...          ...   
1146        1146              356   1 01  1562   
1147        1147              585  18 06  1565   
1148        1148              839  29 08  1570   
1149        1149              993   6 06  1572   
1150        1150               28   6 07  1536   

                                           Intestazione  \
0     VINCENZO BORGHINI IN FIRENZE A GIORGIO VASARI ...   
1     GIORGIO VASARI IN FIRENZE A ORDINE DI SANTO ST...   
2     GABRIEL FIAMMA IN PADOVA A GIORGIO VASARI IN F...   
3     GIORGIO VASARI IN FIRENZE AI DEPUTATI ALLE FAB...   
4     GUGLIELMO SANGALLETTI IN ROMA A GIORGIO VASARI...   
...                                                 ...   
1146           GIORGIO VASARI IN FIRENZE A JACOPO GUIDI   
1147  GIORGIO VASARI IN FIRENZE A VINCENZO BORGHINI ...   
1148     VINCENZO BORGHINI IN POPPIANO A GIORGIO VASARI   
1149  BARTOLOMMEO BUSSOTTO IN ROMA A GIORGIO VASARI ...   
1150  GIORGIO VASARI IN AREZZO A NICCOLÒ SERGUIDI IN...   

                                              Segnatura  \
0                    ASA, AV, 14 (XLVIII), cc. 146-148.   
1                              ASPI, Carte Bonaini, IX.   
2                      ASA, AV, 13 (XLVII), cc. 53, 59.   
3     AAA, filza CC, Registro di lettere, i, 1569, I...   
4                     ASA, AV, 15 (XLIX), cc. 174, 177.   
...                                                 ...   
1146                          ASF, Carte Guidi, c. 356.   
1147                            ASF, CdA, II, I, n. 30.   
1148                       ASA, AV, 14 (XLVIII), c. 70.   
1149                    ASA, AV, 10 (XLIV), cc. 83, 86.   
1150  BRF, 2354, c. 26v, copia di Giorgio Vasari il ...   

                                                  Fonte  \
0     Molto Magnifico messer Giorgio. Ho ricevuto la...   
1     Illustrissimi Signori miei observandissimi. Co...   
2     Magnifico Signor mio osservandissimo. Ho scrit...   
3     Molto magnifici Signori Deputati, Signori mia ...   
4     Molto Magnifico Signor Cavaliere. La presente ...   
...                                                 ...   
1146  Reverendissimo monsignore patron mio, io mando...   
1147  Reverendo Signor mio. S’è riceuto la vostra le...   
1148  Messer Giorgio mio. Io vi mando una lunga dice...   
1149  Molto magnifico Signor come fratello. Scrissi ...   
1150  A messer Niccolo Serguidi sopra l’ornamento di...   

                                           Bibliografia    author  
0                               Frey 1930, pp. 736-737.  Borghini  
1                               Frey 1930, pp. 476-477.    Vasari  
2                               Frey 1923, pp. 513-514.     other  
3                              Frey 1930, pp. 810-811.     Vasari  
4                               Frey 1930, pp. 599-600.     Sanga  
...                                                 ...       ...  
1146  Palli D’Addario 1985, pp. 379-380 (post 2 giug...    Vasari  
1147   Gaye 1839-1840, III, p. 185; Milanesi 1878-18...    Vasari  
1148                            Frey 1930, pp. 522-523.  Borghini  
1149                            Frey 1930, pp. 681-682.     other  
1150  Bottari-Ticozzi 1822-1825, III, pp. 63-67 (6 l...  Vasari_c  

[1151 rows x 8 columns]

In [76]:
dataframe.to_csv(data_path + 'letters_4.csv', index = False, header=True)
#df_labeled = pd.read_csv(data_path + 'all_letters_labeled_author.csv')
dataframe

Unnamed: 0  Numero d'ordine         Data  \
0              0             1051  26 12  1572   
1              1              791  24 12  1569   
2              2              284  20 04  1559   
3              3             1114   1 08  1573   
4              4              904   3 09  1571   
...          ...              ...          ...   
1146        1146              356   1 01  1562   
1147        1147              585  18 06  1565   
1148        1148              839  29 08  1570   
1149        1149              993   6 06  1572   
1150        1150               28   6 07  1536   

                                           Intestazione  \
0     VINCENZO BORGHINI IN FIRENZE A GIORGIO VASARI ...   
1     GIORGIO VASARI IN FIRENZE A ORDINE DI SANTO ST...   
2     GABRIEL FIAMMA IN PADOVA A GIORGIO VASARI IN F...   
3     GIORGIO VASARI IN FIRENZE AI DEPUTATI ALLE FAB...   
4     GUGLIELMO SANGALLETTI IN ROMA A GIORGIO VASARI...   
...                                                 ...   
1146           GIORGIO VASARI IN FIRENZE A JACOPO GUIDI   
1147  GIORGIO VASARI IN FIRENZE A VINCENZO BORGHINI ...   
1148     VINCENZO BORGHINI IN POPPIANO A GIORGIO VASARI   
1149  BARTOLOMMEO BUSSOTTO IN ROMA A GIORGIO VASARI ...   
1150  GIORGIO VASARI IN AREZZO A NICCOLÒ SERGUIDI IN...   

                                              Segnatura  \
0                    ASA, AV, 14 (XLVIII), cc. 146-148.   
1                              ASPI, Carte Bonaini, IX.   
2                      ASA, AV, 13 (XLVII), cc. 53, 59.   
3     AAA, filza CC, Registro di lettere, i, 1569, I...   
4                     ASA, AV, 15 (XLIX), cc. 174, 177.   
...                                                 ...   
1146                          ASF, Carte Guidi, c. 356.   
1147                            ASF, CdA, II, I, n. 30.   
1148                       ASA, AV, 14 (XLVIII), c. 70.   
1149                    ASA, AV, 10 (XLIV), cc. 83, 86.   
1150  BRF, 2354, c. 26v, copia di Giorgio Vasari il ...   

                                                  Fonte  \
0     Molto Magnifico messer Giorgio. Ho ricevuto la...   
1     Illustrissimi Signori miei observandissimi. Co...   
2     Magnifico Signor mio osservandissimo. Ho scrit...   
3     Molto magnifici Signori Deputati, Signori mia ...   
4     Molto Magnifico Signor Cavaliere. La presente ...   
...                                                 ...   
1146  Reverendissimo monsignore patron mio, io mando...   
1147  Reverendo Signor mio. S’è riceuto la vostra le...   
1148  Messer Giorgio mio. Io vi mando una lunga dice...   
1149  Molto magnifico Signor come fratello. Scrissi ...   
1150  A messer Niccolo Serguidi sopra l’ornamento di...   

                                           Bibliografia    author  
0                               Frey 1930, pp. 736-737.  Borghini  
1                               Frey 1930, pp. 476-477.    Vasari  
2                               Frey 1923, pp. 513-514.     other  
3                              Frey 1930, pp. 810-811.     Vasari  
4                               Frey 1930, pp. 599-600.     Sanga  
...                                                 ...       ...  
1146  Palli D’Addario 1985, pp. 379-380 (post 2 giug...    Vasari  
1147   Gaye 1839-1840, III, p. 185; Milanesi 1878-18...    Vasari  
1148                            Frey 1930, pp. 522-523.  Borghini  
1149                            Frey 1930, pp. 681-682.     other  
1150  Bottari-Ticozzi 1822-1825, III, pp. 63-67 (6 l...  Vasari_c  

[1151 rows x 8 columns]

#### make single dataframes

In [63]:
columns = ["Numero d'ordine", "Data", "Intestazione", "Segnatura", "Fonte", "Bibliografia", 'author']

In [64]:
df_Giambullari = pd.DataFrame(Giambullari, columns = columns)
df_Borghini = pd.DataFrame(Borghini, columns = columns)
df_Bartoli = pd.DataFrame(Bartoli, columns = columns)
df_Minerbetti = pd.DataFrame(Minerbetti, columns = columns)
df_Sanga = pd.DataFrame(Sanga, columns = columns)
df_Vasari = pd.DataFrame(Vasari, columns = columns)
df_Vasari_c = pd.DataFrame(Vasari_c, columns = columns)

## save dataframes as csv files

In [42]:
data_path = 'data/results/Vasari_letters_csv/'

In [40]:
df_Giambullari.to_csv(data_path + 'Giambullari.csv', index = False, header=True)
df_Borghini.to_csv(data_path + 'Borghini.csv', index = False, header=True)
df_Bartoli.to_csv(data_path + 'Bartoli.csv', index = False, header=True)
df_Minerbetti.to_csv(data_path + 'Minerbetti.csv', index = False, header=True)
df_Sanga.to_csv(data_path + 'Sanga.csv', index = False, header=True)

### open csv as dataframes

In [43]:
Gia = 'Giambullari.csv'
Bor = 'Borghini.csv'
Min = 'Minerbetti.csv'
Bar = 'Bartoli.csv'
San = 'Sanga.csv'

df_Giambullari = pd.read_csv(data_path+Gia)
df_Borghini = pd.read_csv(data_path+Bor)
df_Bartoli = pd.read_csv(data_path+Bar)
df_Minerbetti = pd.read_csv(data_path+Min)
df_Sanga = pd.read_csv(data_path+San)

# make files for DANTE code and STYLOMETRY

for this GitHub Repo (https://github.com/AlexMoreo/dante-verification) 
the documents must be named "author_texname.txt"

also usefull for "Introduction to stylometry with Python", by  François Dominic Laramée, The Programming Historian 7 (2018), https://doi.org/10.46430/phen0078.

### open and save files

In [90]:
filename = 'data/results/Vasari_letters_csv/letters_4.csv'
dataframe = pd.read_csv(filename) # open csv
#dataframe.to_csv(filename, index = False, header = True) # save as csv

#dataframe

In [ ]:
Dante_path = 'data/results/Dante/'

In [ ]:
# make all letters from dataframe format 'author_text.txt'

for i in range(len(dataframe)):
        
     # get text, letter-number and create filename
    text = df["Fonte"][i] 
    number = str(df["Numero d'ordine"][i]) 
    filename = Dante_path + df['author'][i] + '_' + number + '.txt'
    
     # write .txt file
    output = open(filename, "wt")
    output.write(text)
    output.close()

### make unkown_files

In [50]:
source_path = 'data/Italian-version/file-per-artist-italian/'
unknown_path = 'data/results/Dante_unkown/'

#### make new filenames

In [63]:
list_of_files = os.listdir('data/Italian-version/file-per-artist-italian')
renamed_list = list()

for i in list_of_files:
    renamed_list.append(unknown_path + 'unkown_' + (i.replace(' ', '-').replace("'", "").replace(",","-").strip('_').split('_'))[1] + '.txt')

    

#len(renamed_list), len(list_of_files)
#list_of_files

len(renamed_list), len(list_of_files), renamed_list[7]

(168, 168, 'data/results/Dante_unkown/unkown_Francesco-Francia.txt')

In [56]:
# load file
as_input = open((source_path + list_of_files[0]), 'r')
# create filename
filename = renamed_list[0] 
# take text
#text = as_input.read()
 # write .txt file
#output = open(filename, "wt")
#output.write(text)
#+output.close()
filename
#as_input

'data/results/Dante_unkown/unkown_Lorenzo Costa.txt'

In [64]:
# build iterator for all biographies
index = 0
for i in list_of_files:
    # open file
    as_input = open((source_path + i), 'r')
    # take text
    text = as_input.read()
    as_input.close
    
    output = open(renamed_list[index], "wt")
    output.write(text)
    output.close()
    print(renamed_list[index])
    index += 1
    
    # write .txt file

data/results/Dante_unkown/unkown_Lorenzo-Costa.txt
data/results/Dante_unkown/unkown_Taddeo-Gaddi.txt
data/results/Dante_unkown/unkown_Mariotto-Albertinelli.txt
data/results/Dante_unkown/unkown_Fra-GiovannAgnolo-Montorsoli.txt
data/results/Dante_unkown/unkown_Giovanfrancesco-Rustichi.txt
data/results/Dante_unkown/unkown_Ridolfo--Davit-e-Benedetto-Grillandai.txt
data/results/Dante_unkown/unkown_GiovannAntonio-Lappoli.txt
data/results/Dante_unkown/unkown_Francesco-Francia.txt
data/results/Dante_unkown/unkown_Auteur.txt
data/results/Dante_unkown/unkown_Agostino-et-Agnolo.txt
data/results/Dante_unkown/unkown_Niccolò-detto-il-Tribolo.txt
data/results/Dante_unkown/unkown_Marco-Calavrese.txt
data/results/Dante_unkown/unkown_Francesco-Mazzuoli.txt
data/results/Dante_unkown/unkown_Giovanni-Antonio-Sogliani.txt
data/results/Dante_unkown/unkown_Masolino.txt
data/results/Dante_unkown/unkown_Duccio.txt
data/results/Dante_unkown/unkown_Giovan-Francesco-detto-il-Fattore-e-Pellegrino-da-Modana.txt
data

In [2]:
#os.listdir('data/results/Dante/')

### Helper Funcitons

In [104]:
# obo.py
# get rid of tags without Beautiful Soup
# because there is a space " " missing after some <br/> tags

def stripTags(pageContent):
    pageContents = str(pageContent)

    inside = 0
    text = ''

    for char in pageContents:
        if char == '<':
            inside = 1
        elif (inside == 1 and char == '>'):
            inside = 0
            #after = 1
            #text += " "
        elif inside == 1:
            continue
        else:
            text += char

    return text